client/state/data-getters/index.js

```diff
@@ -82,6 +82,47 @@ export const requestGeoLocation = () =>
 		{ fromApi: () => ( { body: { country_short } } ) => [ [ 'geo', country_short ] ] }
 	);
 
+export const requestRecentPostViews = ( siteId, postIds ) => {
+	const key = `recent-post-views-${ siteId }`;
+	const freshness = 5 * 60 * 1000; // 5 min
+
+	// Don't re-request recent post views that are cached and up to date OR
+	// have a pending request.
+	const freshPostIds = postIds.filter(
+		postId =>
+			getHttpData( `${ key }-${ postId }` ).lastUpdated >= Date.now() - freshness &&
+			getHttpData( `${ key }-${ postId }` ).status !== 'pending'
+	);
+	const requestPostIds = difference( postIds, freshPostIds );
+
+	// Break post_ids into chunks of 100 because `stats/views/posts`
+	// is limited to 100 post_ids per query.
+	return chunk( requestPostIds, 100 ).map( postIdsChunk => {
```

Comment:
as a side-effect of `httpData`'s architecture you should be able to eliminate `chunk()` here too

imagine for a moment the difference between issuing multiple calls here vs. telling Calypso you want all of them

_multiple requests_

we want 400 counts so we issue four API calls. as each comes back in the UI updates. the first batch to return might be 300-400. the second batch comes in and the UI updates, the third batch etc…

each time it updates it still requires the same data, so the `request()` is performed but it sees that all requests are `pending` and so essentially does nothing

_another approach_

now let's skip chunking and just limit this to what the API can handle. say we only grab 100 at a time. the UI loads, requires 400 counts and makes a `request()`, well the request grabs the first 100 and ignores the rest. once the API call returns the UI updates to show the first 100 but it still needs 100-400 so it makes the `request()` again. the `request` sees that 1-100 are fresh enough and so elides them, then grabs the next 100, a second network call is made, and so on.

---

by doing this we can force a serial loading sequence (TCP won't guarantee sequence so if we even ever want to we have to impose that in the client anyway) and also mitigate against sending out so many requests all at the same time. granted, if the `render()` is called quickly in succession it will still issue all requests simultaneously and we're back to the original behavior, except our code is slightly smaller and simpler.

```js
postIdsToRefresh.slice( 0, 100 ).map( … )
```